core/src/main/java/org/elasticsearch/Build.java

```diff
@@ -35,65 +33,51 @@
     public static final Build CURRENT;
 
     static {
-        String hash = "NA";
-        String hashShort = "NA";
-        String timestamp = "NA";
-
-        try (InputStream is = Build.class.getResourceAsStream("/es-build.properties")){
-            Properties props = new Properties();
-            props.load(is);
-            hash = props.getProperty("hash", hash);
-            if (!hash.equals("NA")) {
-                hashShort = hash.substring(0, 7);
-            }
-            String gitTimestampRaw = props.getProperty("timestamp");
-            if (gitTimestampRaw != null) {
-                timestamp = ISODateTimeFormat.dateTimeNoMillis().withZone(DateTimeZone.UTC).print(Long.parseLong(gitTimestampRaw));
-            }
-        } catch (Exception e) {
+        String change = "NA";
+        String buildDate = "NA";
+
+        String path = Build.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+        try {
+            JarFile jar = new JarFile(path);
+            Manifest manifest = jar.getManifest();
+            change = manifest.getMainAttributes().getValue("Change");
+            buildDate = manifest.getMainAttributes().getValue("Build-Date");
+        } catch (IOException e) {
             // just ignore...
         }
 
-        CURRENT = new Build(hash, hashShort, timestamp);
+        CURRENT = new Build(change, buildDate);
     }
 
-    private String hash;
-    private String hashShort;
-    private String timestamp;
-
-    Build(String hash, String hashShort, String timestamp) {
-        this.hash = hash;
-        this.hashShort = hashShort;
-        this.timestamp = timestamp;
-    }
+    private String change;
+    private String buildDate;
 
-    public String hash() {
-        return hash;
+    Build(String hashShort, String timestamp) {
```

Comment:
nit: this is normally called the "short hash" so reversing it sounds very odd. :)